docs/products/postgresql/concepts/upgrade-failover.rst

```diff
@@ -57,3 +57,32 @@ During maintenance updates, cloud migrations, or plan changes, the below procedu
 
 .. Note::
     The old primary server is kept alive for a short period of time (minimum 60 seconds) with a TCP forwarding setup pointing to the new primary server allowing clients to connect before learning the new IP address.
+
+Recreation of replication slots
+-------------------------------
+
+In case of failover or controlled switchover of an Aiven for PostgreSQL service, the replication slots from the old primary server are automatically recreated in the new primary server.
+
+.. important::
+
+    The recreation of replication slots feature is enabled automatically and doesn't require restarting the nodes.
+
+One-node cluster
+""""""""""""""""
+
+Before replacing a node in the one-node cluster, the new node acquires information on replication slots on the original service, re-creates them, and only then the failover is performed.
+
+Multi-node cluster
+""""""""""""""""""
+
+For multi-node setups, replication slots from the primary are synchronized to the standbys periodically. At regular time intervals
+
+* Dependencies for newly-created slots are installed in the corresponding databases (currently, every 30 seconds)
```

Comment:
This sentence without a context is a bit confusing, I would add a bit more.
When the new slot is created on a database, in order to re-create this slot on a standby, we use functionality from `aiven_extras` extension, which needs to be installed in that database. Therefore every 30 seconds there is a job checking that this extension is installed on the databases with logical replication slots.